component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/handler/DPoPAuthenticationHandler.java

```diff
@@ -0,0 +1,220 @@
+                                (requestDTO);
+                OAuth2TokenValidationResponseDTO responseDTO = clientApplicationDTO.getAccessTokenValidationResponse();
+
+                getAuthenticationResult(authenticationResult, responseDTO, authorizationHeader,
+                        authenticationRequest);
+
+                String consumerKey = clientApplicationDTO.getConsumerKey();
+
+                authenticationResult.setAuthenticationStatus(AuthenticationStatus.SUCCESS);
+
+                if (StringUtils.isNotEmpty(responseDTO.getAuthorizedUser())) {
+                    User user = new User();
+                    String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername
+                            (responseDTO.getAuthorizedUser());
+                    user.setUserName(UserCoreUtil.removeDomainFromName(tenantAwareUsername));
+                    user.setUserStoreDomain(UserCoreUtil.extractDomainFromName(tenantAwareUsername));
+                    user.setTenantDomain(MultitenantUtils.getTenantDomain(responseDTO.getAuthorizedUser()));
+                    authenticationContext.setUser(user);
+                }
+
+                authenticationContext.addParameter(DPoPConstants.CONSUMER_KEY, consumerKey);
+                authenticationContext.addParameter(Constants.OAUTH2_ALLOWED_SCOPES, responseDTO.getScope());
+                authenticationContext.addParameter(Constants.OAUTH2_VALIDATE_SCOPE,
+                        AuthConfigurationUtil.getInstance().isScopeValidationEnabled());
+                String serviceProvider = null;
+
+                try {
+                    serviceProvider =
+                            OAuth2Util.getServiceProvider(consumerKey).getApplicationName();
+                } catch (IdentityOAuth2Exception e) {
+                    String error = String.format("Error occurred while getting the Service Provider" +
+                            " by Consumer key: %s", consumerKey);
+                    log.error(error, e);
+                }
+
+                String serviceProviderTenantDomain = null;
+                try {
+                    serviceProviderTenantDomain =
+                            OAuth2Util.getTenantDomainOfOauthApp(consumerKey);
+                } catch (InvalidOAuthClientException | IdentityOAuth2Exception e) {
+
+                    String error = String.format("Error occurred while getting the OAuth App" +
+                            " tenantDomain by Consumer key: %s", consumerKey);
+                    log.error(error, e);
+                }
+
+                if (serviceProvider != null) {
+                    authenticationContext.addParameter(DPoPConstants.SERVICE_PROVIDER, serviceProvider);
+                    if (serviceProviderTenantDomain != null) {
+                        authenticationContext.addParameter(DPoPConstants.SERVICE_PROVIDER_TENANT_DOMAIN,
+                                serviceProviderTenantDomain);
+                    }
```

Comment:
Check and see if `serviceProvider` or `serviceProviderTenantDomain` can ever be null, if not get rid of these conditions.

Even if they could be null, this check doesn't do much about it than not setting the param.

Reply:
Fixed with [260a195](https://github.com/kalanika/identity-oauth-addons/commit/260a195559b47b115eac049a41977336d2c813b2).